### PR TITLE
Workaround for unsupported task URL hook prefix.

### PR DESCRIPTION
**Pull request by menivaitsi**
Build: https://ocd.appscale.com:8080/job/Daily%20Build/1674/

### CODE REVIEW

File: AppTaskQueue/templates/task.py [hunk 1]
```diff
@@ -69,12 +69,19 @@ def get_wait_time(retries, args):
       celery.control.revoke(QUEUE_NAME.request.id)
       return
 
+    # For tasks generated by mapreduce, or destined to be run by a module,
+    # the hostname may have a prefix that corresponds to a different subdomain.
+    # AppScale does not support that type of routing currently, so the main
+    # loadbalancer IP is used here for the execution of a task.
```
Comment (cdonati) [inline AppTaskQueue/templates/task.py]: Would it make sense for the AppServer to be responsible for providing the right hostname (or IP address)?

File: AppTaskQueue/templates/task.py [hunk 2]
```diff
@@ -59,8 +59,8 @@ def get_wait_time(retries, args):
 
     if QUEUE_NAME.request.retries >= args['max_retries'] and \
            args['max_retries'] != 0:
-      logger.error("Task %s with id %s has exceeded retries: %s" % \
-                   (args['task_name'], QUEUE_NAME.request.id, args['max_retries']))
+      logger.error("Task %s with id %s has exceeded retries: %s" %
+        (args['task_name'], QUEUE_NAME.request.id, args['max_retries']))
```
Comment (cdonati) [inline AppTaskQueue/templates/task.py]: Insert a newline after the opening paren or indent this line to match.

File: AppTaskQueue/templates/task.py [hunk 3]
```diff
@@ -102,7 +109,8 @@ def get_wait_time(retries, args):
       if url.query:
         connection.putheader('content-type', 'application/octet-stream')
       else:
-        connection.putheader('content-type', 'application/x-www-form-urlencoded')
+        connection.putheader('content-type',
+          'application/x-www-form-urlencoded')
```
Comment (cdonati) [inline AppTaskQueue/templates/task.py]: Same comment about indentation.

**Comment by menivaitsi** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1677/

**Comment by menivaitsi** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1688/

**Pull request merged**